indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskRunner.java

```diff
@@ -88,11 +89,17 @@
    */
   void shutdown(String taskid, String reason);
 
-  default void shutdown(String taskid, String reasonFormat, Object... args)
+  default void shutdown(String taskid, String reasonFormat, Object... reasonArgs)
   {
-    shutdown(taskid, StringUtils.format(reasonFormat, args));
+    String reason = (getLogger().isInfoEnabled()) ? StringUtils.format(reasonFormat, reasonArgs) : null;
```

Comment:
better to add some comments here to elaborate why we do this here

Reply:
fixed in 7c563eac03 and fb3cf82664